client/src/components/mixins/output-mixin.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Vue } from 'vue-property-decorator'
+import axios from 'axios'
+
+@Component
+export default class OutputMixin extends Vue {
+  /**
+   * Downloads the specified Name Request output.
+   * @param id The Name Request id.
+   */
+  async downloadOutputs (id: string): Promise<void> {
+    try {
+      const url = `namerequests/${id}/result`
+      const headers = { 'Accept': 'application/pdf' }
+
+      // show spinner since the network calls below can take a few seconds
+      this.$root.$emit('showSpinner', true)
+
+      // Request PDF for specified id
+      const response: any = await axios.get(url, { headers: headers, responseType: 'blob' as 'json' })
+
+      // Create a new blob object with mime-type explicitly set, otherwise only Chrome works
+      const blob = new Blob([response.data], { type: 'application/pdf' })
```

Comment:
I think this code already exists in one of the services/modules. Can you just reuse it?

Reply:
ah interesting! I'll see if i can find this.

Reply:
I see some code that is related to payments for receipts, it is different however. Is this the code you were referring too?

Reply:
That sounds like it. Doesn't it just download a PDF? I would rather see a single method for download a PDF. Maybe the receipt code needs to be refactored to use a common download method?

Reply:
I think if it was just hitting a different endpoint, making it generic could work but the request for outputs was purposely removed from the `/payments` endpoint and payment module. 
Imo, i think we leave payments ( receipts ) standalone, but in the future if we find we need a generic download, we modify this mixin method to handle that, is that ok?

Reply:
OK, but could you move the network code to one of the 2 places where there is other network code?

Or do you feel strongly that we should update the architecture to have separate mixins for separate actions?

Reply:
I don't feel strongly enough about it :) We can move it anywhere, although i prefer the mixin approach. 
What are the options, i don't see a place where it really fits other than the `newRequestModule`?
The requests made from `existing-Request-Display` refer to `newrequestModule`, so i guess that's the most logical.
(Although NOT a new request at this point 😆 )

Reply:
kk, relocated!